server/src/main/java/keywhiz/service/permissions/AlwaysAllowPermissionCheck.java

```diff
@@ -0,0 +1,40 @@
+package keywhiz.service.permissions;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.annotation.Counted;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AlwaysAllowPermissionCheck implements PermissionCheck{
+  private static final Logger logger = LoggerFactory.getLogger(AlwaysAllowPermissionCheck.class);
+  private PermissionCheck delegate;
+  private MetricRegistry metricRegistry;
+
+  @Inject
+  public AlwaysAllowPermissionCheck(MetricRegistry metricRegistry, PermissionCheck delegate) {
+    this.delegate = delegate;
+    this.metricRegistry = metricRegistry;
+  }
+
+  @Counted
+  public boolean isAllowed(KeywhizPrincipal source, String action, Object target) {
+    boolean hasPermission = this.delegate.isAllowed(source, action, target);
+    if (!hasPermission) {
+      denied();
+    }
+    if (hasPermission) {
+      logger.info(String.format("In delegate - %s wants to %s %s returns true", source, action, target));
+    } else {
+      logger.info(String.format("In delegate - %s wants to %s %s returns false", source, action, target));
```

Comment:
Since the class name is already included in the log line, we can probably drop the "In delegate - " prefix. Also, since the sources and targets might not always have super obvious string representations (their types won't be included), it'll be more clear if you format the message using fields like so:

Actor: %s Action: %s Target: %s Result: %s

Something like that. Finally, since hasPermission is a boolean you can use that directly via a fourth %s rather than using an if/else and printing "true" and "false" manually.